unstructured/partition/xlsx.py

```diff
@@ -41,17 +48,17 @@
 @add_chunking_strategy()
 def partition_xlsx(
     filename: Optional[str] = None,
-    file: Optional[Union[IO[bytes], SpooledTemporaryFile]] = None,
+    file: Optional[IO[bytes]] = None,
```

Comment:
Does not `partition_xlsx()` expect to pass `SpooledTemporaryFile`?

Reply:
@christinestraub `SpooledTemporaryFile` implements the `IO[bytes]` abstract interface, so it is accepted by the type-checker as a legitimate type for the file parameter. So this code type-checks clean:
```python
f = SpooledTemporaryFile(1000)
partition_xlsx(file=f)
```
So there's no need to distinguish `SpooledTemporaryFile` here in the parameter type from any other binary file-like type, like say `BytesIO` or what you get back from `open(file_path, "rb")`.

It's only later that we want to get more specific so we can replace an instance of `SpooledTemporaryFile` with `BytesIO` (because `zipfile.open()` is incompatible with `SpooledTemporaryFile` in Python versions before 3.12 or something). And that we do with `isinstance()`.